src/factory/src/api.rs

```diff
@@ -102,11 +102,18 @@ impl TokenFactoryCanister {
         let actor = self.state.borrow().consume_provided_cycles_or_icp(caller);
         let cycles = actor.await?;
 
-        let create_token =
-            self.state
-                .borrow()
-                .factory
-                .create_with_cycles(&get_token_bytecode(), (info,), cycles);
+        let wasm = self
+            .state
+            .borrow()
+            .token_wasm
+            .clone()
+            .expect("token_wasm is not set in token state");
+
+        let create_token = self
+            .state
+            .borrow()
+            .factory
+            .create_with_cycles(&wasm, (info,), cycles);
```

Comment:
At an initial glance I can't quite understand why this was done?
`get_token_bytecode` seem to already do what is done here, could you elaborate on the reason for the change?

On another note: you can skip cloning the wasm here:

```rust
let State {
    token_wasm,
    factory,
    ..
} = &*self.state.borrow();

let wasm = token_wasm.as_ref().expect("some reason this can't fail");
let create_token =
    factory.create_with_cycles(wasm, (info,), cycles);
```

Reply:
`get_token_bytecode` includes bytes from a file while this change uses `token_wasm` state field